unicodetools/src/main/java/org/unicode/text/UCD/TestUnicodeInvariants.java

```diff
@@ -113,7 +110,31 @@ enum Expected {empty, not_empty, irrelevant};
 
     private static PrintWriter out;
 
-    public static void testInvariants(String outputFile, boolean doRange) throws IOException {
+    /**
+     * Fetch a reader for our input data.
+     * @param inputFile if null, read DEFAULT_FILE from classpath
+     * @return BufferedReader
+     * @throws IOException
+     */
+    private static BufferedReader getInputReader(String inputFile) throws IOException {
+        if (inputFile != null) {
+            return FileUtilities.openUTF8Reader(Settings.SRC_UCD_DIR, inputFile);
```

Comment:
FYI  This should work in the build bots. For example, https://github.com/unicode-org/unicodetools/blob/main/.github/workflows/build-jsp.yml#L52 does set `-DUNICODETOOLS_REPO_DIR=$(pwd)`, and Settings.SRC_UCD_DIR is relative to that. We read all source data files relative to that path.

Reply:
should but does not